src/ansys/geometry/core/math/matrix.py

```diff
@@ -129,3 +134,40 @@ def __new__(cls, input: np.ndarray | RealSequence | Matrix = DEFAULT_MATRIX44):
             raise ValueError("Matrix44 should only be a 2D array of shape (4,4).")
 
         return obj
+
+    @classmethod
+    def create_matrix_from_rotation(
+        cls, direction_x: "Vector3D", direction_y: "Vector3D"
+    ) -> "Matrix44":
+        """Matrix44 helper method -- create_matrix_from_rotation."""
+        matrix = cls(
+            [
+                [direction_x.x, direction_x.y, direction_x.z, 0],
+                [direction_y.x, direction_y.y, direction_y.z, 0],
+                [0, 0, 1, 0],
+                [0, 0, 0, 1],
+            ]
+        )
+        return matrix
```

Comment:
Why can't there be a Z direction? A rotation is typically defined in any direction, especially in 3D space. You are assuming a planar rotation with this method (with a frozen Z axis) - meaning you can only change the X Y  ref vectors. 

Rotations in space are defined with either quaternions or euler angles (both the angle values and the order of rotation). Please modify accordingly.